### PR TITLE
fix(packaging): fix lua-curl and curl compat on el9

### PR DESCRIPTION
**Pull request by kduret**
## Description

fix lua-curl and curl compat on el9

**Fixes** MON-20735

## Type of change

- [x] Patch fixing an issue (non-breaking change)
- [ ] New functionality (non-breaking change)
- [ ] Breaking change (patch or feature) that might cause side effects breaking part of the Software

## Target serie

- [ ] 21.10.x
- [ ] 22.04.x
- [ ] 22.10.x
- [ ] 23.04.x
- [x] 23.10.x (master)

### CODE REVIEW

File: .github/workflows/stream-connectors-dependencies.yml [hunk 1]
```diff
@@ -79,19 +79,40 @@ jobs:
         working-directory: dependencies/${{ matrix.lib }}
         shell: bash
 
+      - if: ${{ matrix.lib == 'lua-curl' }}
```
Comment (sc979) [inline .github/workflows/stream-connectors-dependencies.yml:82]:
```suggestion
      - if: matrix.lib == 'lua-curl'
```

Comment (sc979) [reply .github/workflows/stream-connectors-dependencies.yml:82]: using name could help while debugging the logs ;)

File: .github/workflows/stream-connectors-dependencies.yml [hunk 2]
```diff
@@ -79,19 +79,40 @@ jobs:
...
+          conan install --requires=libcurl/8.1.2 --build=missing -g CMakeToolchain --deployer=full_deploy
+          conan install --requires=openssl/3.1.1 --build=missing -g CMakeToolchain --deployer=full_deploy
+          conan install --requires=zlib/1.2.13 --build=missing -g CMakeToolchain --deployer=full_deploy
+
+          sed -i "s#^CURL_LIBS.*#CURL_LIBS=-Lfull_deploy/host/libcurl/8.1.2/Release/x86_64/lib -l:libcurl.a -Lfull_deploy/host/openssl/3.1.1/Release/x86_64/lib -l:libssl.a -l:libcrypto.a -L full_deploy/host/zlib/1.2.13/Release/x86_64/lib -l:libz.a#" Makefile
+          make
+
+          cd ..
+
+          mkdir -p dependencies/${{ matrix.lib }}/${{ matrix.lib }}/
```
Comment (sc979) [inline .github/workflows/stream-connectors-dependencies.yml:106]: why create twice "matrix.lib" folder ?

Comment (kduret) [reply .github/workflows/stream-connectors-dependencies.yml:106]: this was already the structure
first level is the dependency directory in the git
second level is the directory of packaged files

File: .github/workflows/stream-connectors-dependencies.yml (hunk 1 above)
Comment (kduret) [reply .github/workflows/stream-connectors-dependencies.yml:82]: this is currently the main syntax we use in the workflows : https://github.com/centreon/centreon/blob/develop/.github/workflows/web.yml#L95
I think it's better to let it for consistency

Comment (sc979) [reply .github/workflows/stream-connectors-dependencies.yml:82]: in this case, ${{}} are not required as it is a string
adding a name to the step, helps while reading the logs

Comment (kduret) [reply .github/workflows/stream-connectors-dependencies.yml:82]: indeed I will add name

**Review by sc979 — APPROVED** [review]

**Review by tuntoja — APPROVED** [review]

**Review by lpinsivy — APPROVED** [review]

**Pull request merged**